File: app/src/main/java/ua/hospes/nfs/marathon/core/di/module/ActivitiesModule.java
package ua.hospes.nfs.marathon.core.di.module;

import android.app.Activity;

import dagger.Binds;
import dagger.Module;
import dagger.android.ActivityKey;
import dagger.android.AndroidInjector;
import dagger.multibindings.IntoMap;
import ua.hospes.nfs.marathon.core.di.components.activities.MainActivitySubComponent;
import ua.hospes.nfs.marathon.core.di.components.activities.RaceItemDetailActivitySubComponent;
import ua.hospes.nfs.marathon.ui.MainActivity;
import ua.hospes.nfs.marathon.ui.race.detail.RaceItemDetailActivity;

/**
 * @author Andrew Khloponin
 */
@Module(subcomponents = {
        MainActivitySubComponent.class,
        RaceItemDetailActivitySubComponent.class
})
public abstract class ActivitiesModule {
    @Binds
    @IntoMap
    @ActivityKey(MainActivity.class)
    abstract AndroidInjector.Factory<? extends Activity> bindMainActivityInjectorFactory(MainActivitySubComponent.Builder builder);

    @Binds
    @IntoMap
    @ActivityKey(RaceItemDetailActivity.class)
    abstract AndroidInjector.Factory<? extends Activity> bindRaceItemDetailActivityInjectorFactory(RaceItemDetailActivitySubComponent.Builder builder);
}

File: app/src/main/java/ua/hospes/rtm/di/module/ActivitiesModule.kt
package ua.hospes.rtm.di.module

import dagger.Module
import dagger.android.ContributesAndroidInjector
import ua.hospes.rtm.ui.MainActivity
import ua.hospes.rtm.ui.PrivacyActivity
import ua.hospes.rtm.ui.TermsActivity
import ua.hospes.rtm.ui.race.detail.RaceItemDetailActivity

@Module
abstract class ActivitiesModule {
    @ContributesAndroidInjector(modules = [MainActivityFragmentsModule::class])
    internal abstract fun bindMainActivity(): MainActivity

    @ContributesAndroidInjector
    internal abstract fun bindRaceItemDetailActivity(): RaceItemDetailActivity


    @ContributesAndroidInjector
    internal abstract fun bindPrivacyActivity(): PrivacyActivity

    @ContributesAndroidInjector
    internal abstract fun bindTermsActivity(): TermsActivity
}
